addon/utils/selection.js

```diff
@@ -63,24 +62,27 @@ export default {
    * @param {Function} itemComparator - comparator for items
    */
   /* eslint-disable complexity */
-  range (items, selectedItems, item, rangeState, itemComparator) {
+  range (items, selectedItems, item, rangeState, itemComparator, itemKey) {
     // If an anchor isn't set, then set the anchor and exit
     const rangeAnchor = rangeState['anchor']
-    if (isNone(rangeAnchor)) {
+
+    const anchor = items.findIndex(currentItem => itemComparator(currentItem, rangeState['anchor']))
+    // If anchor is -1 then it was a anchor from a previous page that we can not find, so reset
```

Comment:
Anchors are being set and assume the 'items' never changes.
When switching pages in pagination, the 20 items that were in 'items' are no longer present.
So looking for anchor in 20 'items' that no longer exist give you -1 in the new 20 
'items'.  In this case pagination requires you to reset anchors entirely.